### DISCUSSION TITLE
How to use 'fs' the right way to load a file during build?

### DISCUSSION THREAD

**Discussion by Makskow** [parcel-bundler/parcel#7138] [Q&A] [answered]
Hello,

I'm working on a web app, where I read a file with plain text and then provide its content to a component. This is the gist of it:
```
// index.ts
import fs from 'fs'

const text = fs.readFileSync(__dirname + '/fileWithText', 'utf-8')
```

It's working as expected when served on the dev server, however it fails during build with the following error:
@parcel/resolver-default: External dependency "fs" is not declared in package.json.

I'd have assumed the emulated Node API would be used at build-time to inline the file content, as shown here: https://parceljs.org/features/node-emulation/#inlining-fs.readfilesync

What would be the right way to make the build work? Is there anything I actually need to add to package.json dependencies?

**Comment by mischnic** [comment, marked as answer]
> External dependency

Looks like you are in library mode. Probably because of this: https://parceljs.org/getting-started/migration/#package.json%23main

If that doesn't fix it: please post your package.json

**Reply by Makskow** [reply]
That helped, thanks a lot for a quick reply! :)
